harvester_e2e_tests/integration/test_vm_functions.py

```diff
@@ -296,18 +296,19 @@ def test_softreboot(self, api_client, unique_vm_name, wait_timeout):
         assert "AgentConnected" == old_agent['type'], (code, data)
 
         api_client.vms.softreboot(unique_vm_name)
+        # Wait until agent disconnected (leaving OS)
         endtime = datetime.now() + timedelta(seconds=wait_timeout)
         while endtime > datetime.now():
             code, data = api_client.vms.get_status(unique_vm_name)
             if "AgentConnected" not in data['status']['conditions'][-1]['type']:
                 break
             sleep(5)
-
+        # then wait agent connected again (Entering OS)
         endtime = datetime.now() + timedelta(seconds=wait_timeout)
         while endtime > datetime.now():
             code, data = api_client.vms.get_status(unique_vm_name)
-            phase, conds = data['status']['phase'], data['status']['conditions']
-            if "Running" == phase and "AgentConnected" == conds[-1]['type']:
+            phase, conds = data['status']['phase'], data['status'].get('conditions', [{}])
```

Comment:
Just to understand this, why do we need `get` here?

Reply:
I think this is a KubeVirt thing.
When we query out to fetch [VirtualMachineStatus](https://kubevirt.io/api-reference/master/definitions.html#_v1_virtualmachinestatus) - `conditions` are an optional array/list (of [VirtualMachineInstanceCondition](https://kubevirt.io/api-reference/master/definitions.html#_v1_virtualmachineinstancecondition) ) - that might or might not exist.

Reply:
When reboot the VM, VM's conditions will be refresh and update, so sometimes we might get a empty conditions from the API.